kernel/mod_subst.ml

```diff
@@ -345,7 +345,14 @@ let rec map_kn f f' c =
     match kind c with
       | Const kn -> (try f' kn with No_subst -> c)
       | Proj (p,r,t) ->
-          let p' = Projection.map f p in
+          (* TODO: how to do this nicely? *)
+          let f_const (c : Constant.t) =
+            match kind (f' (c, UVars.Instance.empty)) with
+            | Const (c', i) -> assert (UVars.Instance.is_empty i); c'
+            | exception No_subst -> c
+            | _ -> assert false
+          in
```

Comment:
This is a little hacky but I don't seem to have access to a usefully typed function in this context that I could call instead.